Explorer/Assets/Scripts/Global/Dynamic/DynamicWorldContainer.cs

```diff
@@ -203,7 +198,14 @@ async UniTask InitializeContainersAsync(IPluginSettingsContainer settingsContain
                 landscapePlugin
             );
 
-            var archipelagoIslandRoom = new ArchipelagoIslandRoom(staticContainer.CharacterContainer.CharacterObject, staticContainer.WebRequestsContainer.WebRequestController, identityCache, multiPool);
+            var archipelagoIslandRoom = new RenewableArchipelagoIslandRoom(
+                () => new ArchipelagoIslandRoom(
```

Comment:
Do we really want to hide dependencies behind a delegate with captures? 
Because of this hiding `RenewableArchipelagoIslandRoom` looks purpose-less. Is not it better for understanding to see in which domain `RenewableArchipelagoIslandRoom` operates rather than hiding it behind the faceless delegate?

Reply:
We may introduce a new factory to avoid a anonymous capture but I see the creating of new factory for this single purpose use case is overwhelming